crates/net/downloaders/src/bodies/bodies.rs

```diff
@@ -289,24 +291,25 @@ where
                         continue
                     }
 
-                    match requests.last().map(|range| *range.end()) {
-                        // Extend the last range if contiguous
-                        Some(range_end) if range_end + 1 == num => {
-                            let range = requests.pop().unwrap();
-                            requests.push(*range.start()..=num);
+                    match requests.last_mut() {
+                        // Extend the last range if it remains contiguous and request size is under
+                        // limit
+                        Some(range)
+                            if range.end == num &&
+                                range.end.saturating_sub(range.start) < max_request_len =>
+                        {
+                            *range = range.start..num + 1; // exclusive range
                         }
                         // Push the new request range
-                        Some(_) | None => requests.push(num..=num),
+                        _ => requests.push(num..num + 1), // exclusive range
```

Comment:
I understand that ranges are exclusive but it would be helpful to document why we need them inclusive here

Reply:
added a comment